slurm/install/install.py

```diff
@@ -257,11 +287,28 @@ def accounting(s: InstallSettings) -> None:
 
 def complete_install(s: InstallSettings) -> None:
     if s.mode == "scheduler":
-        _complete_install_primary(s)
-    _complete_install_all(s)
+        if s.is_primary_scheduler:
+            _complete_install_primary(s)
+        _complete_install_all(s)
+    else:
+        _complete_install_all(s)
 
 
 def _complete_install_primary(s: InstallSettings) -> None:
+    """
+    Only the primary scheduler should be creating files under /sched.
+    """
+    assert s.is_primary_scheduler
+    secondary_scheduler = None
+    if s.secondary_scheduler_name:
+        secondary_scheduler = ilib.await_node_hostname(
+            s.config, s.secondary_scheduler_name
+        )
+    state_save_location = "/var/spool/slurmd"
```

Comment:
should we be calling scheduler's state save location as "slurmd"? that sounds a bit confusing, could it be "slurmctld"?

Reply:
Good call, slurmctld makes more sense, but we shipped with this in 3.0.3 and in 2.x. Maybe we should create a symlink to /var/spool/slurmd + do a check on their slurm config to tell them this is deprecated?

Reply:
I am not sure about that, i mean i think just fixing it should be enough..The reason is var/spool/slurmd is a valid location that is used for slurmd-- granted slurmd is not running alongside slurmctld so we wont run into issues. I think its not necssary to symlink though because this just reads like a typo or a mistake in porting over configs.